pcs/lib/corosync/qdevice_net.py

```diff
@@ -108,9 +113,154 @@ def set_up_client_certificates(
     run_and_raise(communicator_factory.get_communicator(), com_cmd_4)
 
 
+def setup_client_certificates_local(
+    runner: CommandRunner,
+    reporter: reports.ReportProcessor,
+    communicator_factory: NodeCommunicatorFactory,
+    qnetd_target: RequestTarget,
+    cluster_name: str,
+) -> None:
+    """
+    setup the local node for using qdevice model net
+
+    runner -- command runner instance
+    reporter -- report processor instance
+    communicator_factory -- communicator factory instance
+    qnetd_target -- qdevice provider (qnetd host)
+    cluster_name -- name of the cluster to which qdevice is being added
+    """
+    # get qnetd CA certificate
+    com_cmd_1 = qdevice_net_com.GetCaCert(reporter)
+    com_cmd_1.set_targets([qnetd_target])
+    qnetd_ca_cert = run_and_raise(
+        communicator_factory.get_communicator(), com_cmd_1
+    )[0][1]
+    # init certificate storage on local node
+    client_setup(runner, qnetd_ca_cert)
+    # create client certificate request
+    cert_request = client_generate_certificate_request(runner, cluster_name)
+    # sign the request on qnetd host
+    com_cmd_2 = qdevice_net_com.SignCertificate(reporter)
+    com_cmd_2.add_request(qnetd_target, cert_request, cluster_name)
+    signed_certificate = run_and_raise(
+        communicator_factory.get_communicator(), com_cmd_2
+    )[0][1]
+    # transform the signed certificate to pk12 format which can sent to nodes
+    pk12 = client_cert_request_to_pk12(runner, signed_certificate)
+    # store final certificate
+    client_import_certificate_and_key(runner, pk12)
+
+
+def check_client_certificates_local(
+    runner: CommandRunner,
+    reporter: reports.ReportProcessor,
+    communicator_factory: NodeCommunicatorFactory,
+    qnetd_target: RequestTarget,
+    cluster_name: str,
+) -> bool:
+    """
+    check if the local node has qdevice model net certificates
+
+    runner -- command runner instance
+    reporter -- report processor instance
+    communicator_factory -- communicator factory instance
+    qnetd_target -- qdevice provider (qnetd host)
+    cluster_name -- name of the cluster to which qdevice is being added
+    """
+
+    def get_certificate_data(text: str) -> Optional[str]:
+        match = re.search(
+            r"-----BEGIN CERTIFICATE-----([a-zA-Z0-9/+=]+)-----END CERTIFICATE-----",
+            "".join(text.splitlines()),
+        )
+        return match.group(1) if match else None
+
+    if not client_initialized():
+        return False
+
+    # check if certificates are present
+    certutil_cmd = [
+        settings.certutil_executable,
+        "-d",
+        settings.corosync_qdevice_net_client_certs_dir,
```

Comment:
Option `-L` could be added here at it is added everywhere. And `certutil_cmd` can be `tuple` instead of `list`.